global: ensure unicode handling in all user-facing code

chris-asl opened the pull request:
Closes #2133 

Signed-off-by: Chris Aslanoglou <chris.aslanoglou@gmail.com>

jacquerie commented on tests/unit/utils/test_utils_pubnote.py:
```diff
@@ -0,0 +1,37 @@
+    unicode_title = u'φοο_title'
+    unicode_volume = u'φοο_volume'
+
+    expected = u'{},{}'.format(unicode_title, unicode_volume)
```
Mh, read this: https://testing.googleblog.com/2014/07/testing-on-toilet-dont-put-logic-in.html

chris-asl replied:
Thanks for the link! 
I've updated the construction of `expected`.

kaplun replied:
Indeed I also learned something new today!

michamos commented on inspirehep/utils/conferences.py:
```diff
@@ -53,7 +53,7 @@ def render_conferences_contributions(cnum):
 def conferences_in_the_same_series_from_es(seriesname):
     """Query ES for conferences in the same series."""
-    query = 'series:"{}"'.format(seriesname)
+    query = u'series:"{}"'.format(seriesname)
```
:+1: (sorry, just testing code review workflow with @kaplun and @jacquerie )

kaplun replied:
While you are modifying this query, is there a safer way to escape `seriesname`? I assume that when it will contain a `"`, the call will blow up.

kaplun commented on inspirehep/utils/experiments.py:
```diff
@@ -54,7 +54,7 @@ def render_experiment_people(experiment_name):
 def experiment_contributions_from_es(experiment_name):
     """Query ES for conferences in the same series."""
-    query = 'accelerator_experiments.experiment:"{}"'.format(experiment_name)
+    query = u'accelerator_experiments.experiment:"{}"'.format(experiment_name)
```
I see that also this part would require proper escaping of `experiment_name`.

jacquerie commented:
> LGTM.

This is not done, though (it's still in WIP, and if you grep the codebase you'll find lots of other things to fix).

chris-asl commented:
(Hopefully) closes #2133

jacquerie commented:
> (Hopefully) closes #2133

GitHub is dumb, and understand these comments only in the first comment of the PR with a specific format: https://help.github.com/articles/closing-issues-via-commit-messages/

jmartinm commented:
As an alternative, we can try adding:

```
from __future__ import unicode_literals
```

on top of all files.

This needs to be tested to avoid unexpected side effects.

chris-asl commented:
Resurrected this PR as per @ammirate suggestion. 
Fixed some commit messages and rebased master ontop of the PR branch.

Regarding @kaplun suggestions, [here](https://github.com/inspirehep/inspire-next/pull/2199#discussion_r110407514) and [here](https://github.com/inspirehep/inspire-next/pull/2199#discussion_r110409184), I can integrate them in a another PR.

jacquerie commented:
@krzysztof pointed out in https://github.com/inveniosoftware/invenio/issues/3830#issue-262781413 an interesting tool to check if we actually got all these: https://github.com/cogniteev/flake8-unicode-string-literal

Spoiler alter: we didn't. But I'd say that we should add this dependency!

jacquerie commented:
This became stale, but most of the errors were fixed anyway. In the future we might add `unicode-string-literal` (like in https://github.com/inspirehep/inspire-dojson/commit/c8556d45eb2c16707e1237c0106565cf734d8924) to squash these errors at integration time.
